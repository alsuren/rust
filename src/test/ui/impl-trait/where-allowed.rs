use std::fmt::Debug;
fn in_impl_Fn_return_in_parameters(_: &impl Fn() -> impl Debug) { panic!() }
