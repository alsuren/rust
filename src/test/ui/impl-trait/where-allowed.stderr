error[E0666]: nested `impl Trait` is not allowed
  --> $DIR/where-allowed.rs:50:51
   |
LL | fn in_impl_Fn_parameter_in_parameters(_: &impl Fn(impl Debug)) { panic!() }
   |                                           --------^^^^^^^^^^-
   |                                           |       |
   |                                           |       nested `impl Trait` here
   |                                           outer `impl Trait`

error[E0666]: nested `impl Trait` is not allowed
  --> $DIR/where-allowed.rs:55:57
   |
LL | fn in_impl_Fn_parameter_in_return() -> &'static impl Fn(impl Debug) { panic!() }
   |                                                 --------^^^^^^^^^^-
   |                                                 |       |
   |                                                 |       nested `impl Trait` here
   |                                                 outer `impl Trait`

error[E0658]: `impl Trait` in type aliases is unstable
  --> $DIR/where-allowed.rs:118:16
   |
LL |     type Out = impl Debug;
   |                ^^^^^^^^^^
   |
   = note: see issue #63063 <https://github.com/rust-lang/rust/issues/63063> for more information
   = help: add `#![feature(type_alias_impl_trait)]` to the crate attributes to enable

error[E0658]: `impl Trait` in type aliases is unstable
  --> $DIR/where-allowed.rs:154:23
   |
LL | type InTypeAlias<R> = impl Debug;
   |                       ^^^^^^^^^^
   |
   = note: see issue #63063 <https://github.com/rust-lang/rust/issues/63063> for more information
   = help: add `#![feature(type_alias_impl_trait)]` to the crate attributes to enable

error[E0658]: `impl Trait` in type aliases is unstable
  --> $DIR/where-allowed.rs:158:39
   |
LL | type InReturnInTypeAlias<R> = fn() -> impl Debug;
   |                                       ^^^^^^^^^^
   |
   = note: see issue #63063 <https://github.com/rust-lang/rust/issues/63063> for more information
   = help: add `#![feature(type_alias_impl_trait)]` to the crate attributes to enable

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:15:53
   |
LL | fn in_impl_Fn_return_in_parameters(_: &impl Fn() -> impl Debug) { panic!() }
   |                                                     ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:18:40
   |
LL | fn in_fn_parameter_in_parameters(_: fn(impl Debug)) { panic!() }
   |                                        ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:22:42
   |
LL | fn in_fn_return_in_parameters(_: fn() -> impl Debug) { panic!() }
   |                                          ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:26:38
   |
LL | fn in_fn_parameter_in_return() -> fn(impl Debug) { panic!() }
   |                                      ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:30:40
   |
LL | fn in_fn_return_in_return() -> fn() -> impl Debug { panic!() }
   |                                        ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:34:49
   |
LL | fn in_dyn_Fn_parameter_in_parameters(_: &dyn Fn(impl Debug)) { panic!() }
   |                                                 ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:38:51
   |
LL | fn in_dyn_Fn_return_in_parameters(_: &dyn Fn() -> impl Debug) { panic!() }
   |                                                   ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:42:55
   |
LL | fn in_dyn_Fn_parameter_in_return() -> &'static dyn Fn(impl Debug) { panic!() }
   |                                                       ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:46:57
   |
LL | fn in_dyn_Fn_return_in_return() -> &'static dyn Fn() -> impl Debug { panic!() }
   |                                                         ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:50:51
   |
LL | fn in_impl_Fn_parameter_in_parameters(_: &impl Fn(impl Debug)) { panic!() }
   |                                                   ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:55:57
   |
LL | fn in_impl_Fn_parameter_in_return() -> &'static impl Fn(impl Debug) { panic!() }
   |                                                         ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:60:59
   |
LL | fn in_impl_Fn_return_in_return() -> &'static impl Fn() -> impl Debug { panic!() }
   |                                                           ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:64:38
   |
LL | fn in_Fn_parameter_in_generics<F: Fn(impl Debug)> (_: F) { panic!() }
   |                                      ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:68:40
   |
LL | fn in_Fn_return_in_generics<F: Fn() -> impl Debug> (_: F) { panic!() }
   |                                        ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:81:32
   |
LL | struct InBraceStructField { x: impl Debug }
   |                                ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:85:41
   |
LL | struct InAdtInBraceStructField { x: Vec<impl Debug> }
   |                                         ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:89:27
   |
LL | struct InTupleStructField(impl Debug);
   |                           ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:94:25
   |
LL |     InBraceVariant { x: impl Debug },
   |                         ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:96:20
   |
LL |     InTupleVariant(impl Debug),
   |                    ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:107:23
   |
LL |     fn in_return() -> impl Debug;
   |                       ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:125:34
   |
LL |     fn in_trait_impl_return() -> impl Debug { () }
   |                                  ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:138:33
   |
LL |     fn in_foreign_parameters(_: impl Debug);
   |                                 ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:141:31
   |
LL |     fn in_foreign_return() -> impl Debug;
   |                               ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:158:39
   |
LL | type InReturnInTypeAlias<R> = fn() -> impl Debug;
   |                                       ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:163:16
   |
LL | impl PartialEq<impl Debug> for () {
   |                ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:168:24
   |
LL | impl PartialEq<()> for impl Debug {
   |                        ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:173:6
   |
LL | impl impl Debug {
   |      ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:179:24
   |
LL | impl InInherentImplAdt<impl Debug> {
   |                        ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:185:11
   |
LL |     where impl Debug: Debug
   |           ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:192:15
   |
LL |     where Vec<impl Debug>: Debug
   |               ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:199:24
   |
LL |     where T: PartialEq<impl Debug>
   |                        ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:206:17
   |
LL |     where T: Fn(impl Debug)
   |                 ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:213:22
   |
LL |     where T: Fn() -> impl Debug
   |                      ^^^^^^^^^^

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:219:29
   |
LL |     let _in_local_variable: impl Fn() = || {};
   |                             ^^^^^^^^^
   |
   = help: add `#![feature(impl_trait_in_bindings)]` to the crate attributes to enable

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/where-allowed.rs:221:46
   |
LL |     let _in_return_in_local_variable = || -> impl Fn() { || {} };
   |                                              ^^^^^^^^^

error: could not find defining uses
  --> $DIR/where-allowed.rs:154:1
   |
LL | type InTypeAlias<R> = impl Debug;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: could not find defining uses
  --> $DIR/where-allowed.rs:118:5
   |
LL |     type Out = impl Debug;
   |     ^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 42 previous errors

Some errors have detailed explanations: E0562, E0658, E0666.
For more information about an error, try `rustc --explain E0562`.
